### PR TITLE
#3173 - SA1611 ignores included documentation

### PR DESCRIPTION
**Pull request by angelobreuer**
Tracking: #3173

This PR fixes issue #3173 and adds tests for it.

### CODE REVIEW

File: StyleCop.Analyzers/StyleCop.Analyzers/DocumentationRules/ElementDocumentationBase.cs [hunk 1]
```diff
@@ -235,28 +235,28 @@ private void HandleDeclaration(SyntaxNodeAnalysisContext context, StyleCopSettin
                     .Where(x => !string.Equals(x.GetName()?.ToString(), XmlCommentHelper.IncludeXmlTag, StringComparison.Ordinal))
                 : documentation.Content.GetXmlElements(this.matchElementName);
 
-            if (!matchingXmlElements.Any())
```
Comment (sharwell) [inline StyleCop.Analyzers/StyleCop.Analyzers/DocumentationRules/ElementDocumentationBase.cs:238]: 💭 I'm trying to remember why we have this condition.

File: StyleCop.Analyzers/StyleCop.Analyzers/DocumentationRules/ElementDocumentationBase.cs [hunk 2]
```diff
@@ -235,28 +235,28 @@ private void HandleDeclaration(SyntaxNodeAnalysisContext context, StyleCopSettin
...
+            var isInheritingDocumentation = documentation.Content
+                .Any(x => string.Equals(x.GetName()?.ToString(), XmlCommentHelper.IncludeXmlTag, StringComparison.Ordinal));
```
Comment (sharwell) [inline StyleCop.Analyzers/StyleCop.Analyzers/DocumentationRules/ElementDocumentationBase.cs:239]: ❔ Why not this?

```suggestion
            var isInheritingDocumentation =
                documentation.Content.GetFirstXmlElement(XmlCommentHelper.IncludeXmlTag) is object;
```

Comment (angelobreuer) [reply StyleCop.Analyzers/StyleCop.Analyzers/DocumentationRules/ElementDocumentationBase.cs:239]: Inspired from the code above:
https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/ad20436460c519798c7d48e6aef0b3f4f606e96a/StyleCop.Analyzers/StyleCop.Analyzers/DocumentationRules/ElementDocumentationBase.cs#L235

File: StyleCop.Analyzers/StyleCop.Analyzers/DocumentationRules/ElementDocumentationBase.cs [hunk 3]
```diff
@@ -235,28 +235,28 @@ private void HandleDeclaration(SyntaxNodeAnalysisContext context, StyleCopSettin
                     .Where(x => !string.Equals(x.GetName()?.ToString(), XmlCommentHelper.IncludeXmlTag, StringComparison.Ordinal))
                 : documentation.Content.GetXmlElements(this.matchElementName);
 
-            if (!matchingXmlElements.Any())
+            var isInheritingDocumentation =
+                documentation.Content.GetFirstXmlElement(XmlCommentHelper.IncludeXmlTag) is object;
```
Comment (vweijsters) [inline StyleCop.Analyzers/StyleCop.Analyzers/DocumentationRules/ElementDocumentationBase.cs:239]: I would prefer hasIncludedDocumentation, as the current name could cause confusion with the <inheritdoc/> tag.

File: StyleCop.Analyzers/StyleCop.Analyzers/DocumentationRules/ElementDocumentationBase.cs (hunk 1 above)
Comment (vweijsters) [reply StyleCop.Analyzers/StyleCop.Analyzers/DocumentationRules/ElementDocumentationBase.cs:238]: I don't think there is any specific reason for it, at least not what I could find in the history.

File: StyleCop.Analyzers/StyleCop.Analyzers/DocumentationRules/ElementDocumentationBase.cs [hunk 4]
```diff
@@ -235,28 +235,28 @@ private void HandleDeclaration(SyntaxNodeAnalysisContext context, StyleCopSettin
...
+                    return;
                 }
-            }
 
-            this.HandleXmlElement(context, settings, needsComment, matchingXmlElements, locations);
+                this.HandleCompleteDocumentation(context, needsComment, completeDocumentation, locations);
+            }
+            else
+            {
+                this.HandleXmlElement(context, settings, needsComment, matchingXmlElements, locations);
```
Comment (vweijsters) [inline StyleCop.Analyzers/StyleCop.Analyzers/DocumentationRules/ElementDocumentationBase.cs:258]: Small performance optimization: lines 232-236 can be moved here, as matchingXmlElements is only used here

Comment (angelobreuer) [reply StyleCop.Analyzers/StyleCop.Analyzers/DocumentationRules/ElementDocumentationBase.cs:258]: Thanks! I have commited the changes to 64de3245af819d2a6c716a919fc47e11c2c24a0a.

File: StyleCop.Analyzers/StyleCop.Analyzers/DocumentationRules/ElementDocumentationBase.cs (hunk 3 above)
Comment (angelobreuer) [reply StyleCop.Analyzers/StyleCop.Analyzers/DocumentationRules/ElementDocumentationBase.cs]: Thanks! I have commited the changes to 64de3245af819d2a6c716a919fc47e11c2c24a0a.

**Pull request merged**